pages/common/docker-cp.md

```diff
@@ -0,0 +1,16 @@
+# docker cp
+
+> Copy files or directories between host and container filesystems.
+> More information: <https://docs.docker.com/engine/reference/commandline/cp>.
+
+- Copy a file or directory from the host to a container:
+
+`docker cp {{path/to/file_or_directory_on_host}} {{container_name}}:{{path/to/file_or_directory_in_container}}`
+
+- Copy a file or directory from a container to the host:
+
+`docker cp {{container_name}}:{{path/to/file_or_directory_in_container}} {{path/to/file_or_directory_on_host}}`
+
+- Copy a file or directory from the host to a container, following symlinks (copies the symlinked files directly, not the symlinks themselves):
+
+`docker cp {{path/to/symlink_on_host}} {{container_name}}:{{path/to/file_or_directory_in_container}}`
```

Comment:
Without this flag, this example doesn't work on my machine:
```suggestion
`docker cp --follow-link {{path/to/symlink_on_host}} {{container_name}}:{{path/to/file_or_directory_in_container}}`
```

Reply:
I'll verify on my machine

Reply:
Indeed, `docker cp` docs state that flag is required for that use case (which I haven't needed/used ever so deliberately didn't include but I'm happy for others to work it out given all I need to do is hit a button to commit the suggestions).

Reply:
So this is what I get when using symlinks (`ln -s`):
```console
$ mkdir test_link
$ cd test_link
$ touch file
$ ln -s file link
$ ll
total 0
drwxr-xr-x    4 me staff  128 Mar 19 08:45 ./
drwxr-xr-x+ 271 me staff 8672 Mar 19 08:45 ../
-rw-r--r--    1 me staff    0 Mar 19 08:45 file
lrwxr-xr-x    1 me staff    4 Mar 19 08:45 link -> file
$ docker create ubuntu tail -f /dev/null
d4a72d217cf510d1977539fb7a28156ab00da8ed1a6cf58a3b6d65ca115e0c7e
$ docker cp link d4a72d21:/
$ docker start d4a72d21
$ docker exec d4a72d21 ls -la /
d4a72d21
$ docker exec d4a72d21 ls -la /
total 56
drwxr-xr-x   1 root root    4096 Mar 19 12:45 .
drwxr-xr-x   1 root root    4096 Mar 19 12:45 ..
-rwxr-xr-x   1 root root       0 Mar 19 12:45 .dockerenv
lrwxrwxrwx   1 root root       7 Jan 19 01:01 bin -> usr/bin
drwxr-xr-x   2 root root    4096 Apr 15  2020 boot
drwxr-xr-x   5 root root     340 Mar 19 12:45 dev
drwxr-xr-x   1 root root    4096 Mar 19 12:45 etc
drwxr-xr-x   2 root root    4096 Apr 15  2020 home
lrwxrwxrwx   1 root root       7 Jan 19 01:01 lib -> usr/lib
lrwxrwxrwx   1 root root       9 Jan 19 01:01 lib32 -> usr/lib32
lrwxrwxrwx   1 root root       9 Jan 19 01:01 lib64 -> usr/lib64
lrwxrwxrwx   1 root root      10 Jan 19 01:01 libx32 -> usr/libx32
lrwxrwxrwx   1  501 dialout    4 Mar 19 12:45 link -> file
drwxr-xr-x   2 root root    4096 Jan 19 01:01 media
drwxr-xr-x   2 root root    4096 Jan 19 01:01 mnt
drwxr-xr-x   2 root root    4096 Jan 19 01:01 opt
dr-xr-xr-x 179 root root       0 Mar 19 12:45 proc
drwx------   2 root root    4096 Jan 19 01:04 root
drwxr-xr-x   1 root root    4096 Jan 21 03:38 run
lrwxrwxrwx   1 root root       8 Jan 19 01:01 sbin -> usr/sbin
drwxr-xr-x   2 root root    4096 Jan 19 01:01 srv
dr-xr-xr-x  13 root root       0 Mar 19 12:38 sys
drwxrwxrwt   2 root root    4096 Jan 19 01:04 tmp
drwxr-xr-x   1 root root    4096 Jan 19 01:01 usr
drwxr-xr-x   1 root root    4096 Jan 19 01:04 var
```

Reply:
Here's the results with hard links (`ln <src> <dest>`):
**tl;dr:** it succeeded
```console
$ mkdir test_hard_link
$ cd test_hard_link
$ touch file
$ ln file link
$ ll
total 0
drwxr-xr-x    4 me staff  128 Mar 19 08:50 ./
drwxr-xr-x+ 272 me staff 8704 Mar 19 08:50 ../
-rw-r--r--    2 me staff    0 Mar 19 08:50 file
-rw-r--r--    2 me staff    0 Mar 19 08:50 link
$ docker create ubuntu tail -f /dev/null
2de2c71eb56f4ba9e335637cc01f2550f119573dd7a2410e2ea3b35c36d6be38
$ docker start 2de2c71
2de2c71
$ docker cp link 2de2c71:/
$ docker exec 2de2c71 ls -la / | grep link
-rw-r--r--   1  501 dialout    0 Mar 19 12:50 link
```

Reply:
@bl-ue unless I'm misunderstanding something (definitely possible!) that shows it successfully copied the link, not the file which is what the example is wanting to show.

Reply:
**tl;dr:** copying a symlink and a hard link with `docker cp` and **without** `--follow-link` worked for me.  @navarroaxel I'm not sure what your configuration is.

This is mine:
```console
$ docker version
Client: Docker Engine - Community
 Cloud integration: 1.0.9
 Version:           20.10.5
 API version:       1.41
 Go version:        go1.13.15
 Git commit:        55c4c88
 Built:             Tue Mar  2 20:13:00 2021
 OS/Arch:           darwin/amd64
 Context:           default
 Experimental:      true

Server: Docker Engine - Community
 Engine:
  Version:          20.10.5
  API version:      1.41 (minimum version 1.12)
  Go version:       go1.13.15
  Git commit:       363e9a8
  Built:            Tue Mar  2 20:15:47 2021
  OS/Arch:          linux/amd64
  Experimental:     false
 containerd:
  Version:          1.4.3
  GitCommit:        269548fa27e0089a8b8278fc4fc781d7f65a939b
 runc:
  Version:          1.0.0-rc92
  GitCommit:        ff819c7e9184c13b7c2607fe6c30ae19403a7aff
 docker-init:
  Version:          0.19.0
  GitCommit:        de40ad0
```

Reply:
I used a symlink folder, the content of the **folder** `nnn` was copied only with the `--follow-links` argument:
```
$ ls -lsh
plugins -> /home/.../nnn
$ docker version
Client:
 Version:           20.10.5
 API version:       1.41
 Go version:        go1.16
 Git commit:        55c4c88966
 Built:             Wed Mar  3 16:51:54 2021
 OS/Arch:           linux/amd64
 Context:           default
 Experimental:      true

Server:
 Engine:
  Version:          20.10.5
  API version:      1.41 (minimum version 1.12)
  Go version:       go1.16
  Git commit:       363e9a88a1
  Built:            Wed Mar  3 16:51:28 2021
  OS/Arch:          linux/amd64
  Experimental:     false
 containerd:
  Version:          v1.4.4
  GitCommit:        05f951a3781f4f2c1911b05e61c160e9c30eaa8e.m
 runc:
  Version:          1.0.0-rc93
  GitCommit:        12644e614e25b05da6fd08a38ffa0cfe1903fdec
 docker-init:
  Version:          0.19.0
  GitCommit:        de40ad0
```

Reply:
>@bl-ue unless I'm misunderstanding something (definitely possible!) that shows it successfully copied the link, not the file which is what the example is wanting to show.

Yeah, looks like I misread the issue 😂 

Then forget all this, and apply the suggestion @phrohdoh 👍🏻